docs/core/metrics_datadog.md

```diff
@@ -0,0 +1,235 @@
+---
+title: Datadog
+description: Core utility
+---
+<!-- markdownlint-disable MD013 -->
+Datadog provider creates custom metrics by flushing metrics to standard output and exporting metrics using [Datadog Forwarder](https://docs.datadoghq.com/logs/guide/forwarder/?tab=cloudformation){target="_blank" rel="nofollow"} or flushing metrics to [Datadog extension](https://docs.datadoghq.com/serverless/installation/python/?tab=datadogcli){target="_blank" rel="nofollow"} using Datadog SDK.
+<!-- markdownlint-enable MD013 -->
+
```

Comment:
add a mermaid diagram demonstrating the flow so people can see it upfront what this is